web/src/components/Pagination.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+
+import styled, { DefaultTheme } from 'styled-components';
+
+const PaginationContainer = styled.nav`
+  text-align: center;
+`;
+
+const ButtonList = styled.ul`
+  text-align: center;
+  padding: 1px;
+`;
+
+const Button = styled.li`
+  display: inline-flex;
+  font-family: 'Google Sans', Roboto, Arial, sans-serif;
+  font-size: 13px;
+  font-weight: 700;
+  align-items: center;
+  border-radius: 10px;
+  border-style: none;
+  box-sizing: border-box;
+  height: 25px;
+  padding: 2px 18px;
+  margin: 5px;
+  transition: box-shadow 280ms cubic-bezier(0.4, 0, 0.2, 1), opacity 15ms linear 30ms,
+    transform 270ms cubic-bezier(0, 0, 0.2, 1) 0ms;
+  color: ${(props) => props.theme.colors.default};
+  cursor: pointer;
+`;
+
+const PageButton = styled(Button)`
+  background-color: ${(props) => props.theme.colors.button};
+
+  &:hover {
+    background-color: ${(props) => props.theme.colors.btnhover};
+  }
+`;
+
+const SelectedPageButton = styled(Button)`
+  background-color: ${(props) => props.theme.colors.header};
+
+  &:hover {
+    background-color: ${(props) => props.theme.colors.header};
+  }
+`;
+
+const ArrowButton = styled(Button)`
+  background-color: ${(props) => props.theme.colors.container};
+
+  &:hover {
+    background-color: ${(props) => props.theme.colors.header};
+  }
+`;
+
+interface PaginationProps {
+  currentPage: number;
+  numOfTotalPage: number;
+  numOfPageToShow: number;
+  setPage: React.Dispatch<React.SetStateAction<number>>;
+  theme: DefaultTheme;
+}
+
+export default function Pagination({ currentPage, numOfTotalPage, numOfPageToShow, setPage, theme }: PaginationProps) {
+  const [pageListIndex, setPageListIndex] = useState<number>(Math.floor((currentPage - 1) / numOfPageToShow));
+  const pageListLength = Math.floor(numOfTotalPage / numOfPageToShow);
+  const pageIndex = [];
+
+  for (let i = 1; i <= numOfTotalPage; i += 1) {
+    pageIndex.push(i);
+  }
```

Comment:
To create an array of numbers, check https://stackoverflow.com/a/65244361/17320013 for performance.